document/src/main/java/com/yahoo/document/Document.java

```diff
@@ -266,6 +276,17 @@ public void printXml(XmlStream xml) {
         XmlSerializationHelper.printDocumentXml(this, xml);
     }
 
+    /**
+     * Get JSON representation of the document root and its children contained in a JSON object
+     * @return JSON representation of document
+     */
+    public String toJson() {
+        ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+        JsonWriter writer = new JsonWriter(buffer);
+        writer.write(this);
+        return buffer.toString();
```

Comment:
I'm wondering if we should explicitly use utf-8 as the charset here. Not doing so seems to invoke the "platform's default character set" which is _likely_ to be utf-8, but not guaranteed.

@bratseth what are the best practices here?

Reply:
Yes, you are right: Always send charset explicitly.

Reply:
Good point; fixed in next commit. If only that interface allowed for a Charset object instead of a String lookup with an impossible exception..